src/main/java/org/sonatype/repository/helm/internal/metadata/IndexYamlAbsoluteUrlRewriter.java

```diff
@@ -108,10 +108,9 @@ private Event maybeSetAbsoluteUrlAsRelative(ScalarEvent scalarEvent) throws Malf
       URI uri = new URIBuilder(oldUrl).build();
       if (uri.isAbsolute()) {
         String fileName = uri.getPath();
-        // Strip leading slash
-        if (!fileName.isEmpty()) {
-            fileName = fileName.substring(1);
-        }
+        //remove leading "/" to avoid URI missing, by ref: https://github.com/helm/helm/issues/3878
+        //fix https://github.com/sonatype-nexus-community/nexus-repository-helm/issues/14
+        fileName = fileName.startsWith("/") ? fileName.substring(fileName.lastIndexOf("/") + 1) : fileName;
```

Comment:
Ok! Now I understand the change you are making.

This assumes that you setup a repo that is like:

`http://host.name/path/`

That path has an index.yaml underneath it, and then also the charts, and the charts are being referenced via:

`http://host.name/path/chart.tgz`

This change will make that work for sure, and should work fine for others where the all the files are at the root, however if someone does:

`http://host.name/index.yaml`

`http://host.name/charts/charts.tgz`

It will fail in this case. I'm not knowledgeable enough to know if anyone has done that yet, but we should think about it. Any thoughts?

Reply:
If this is the case, I think I can add a path to the original characters in the metadata. At the time of the proxy, get the original address

Reply:
In fact, what you say is similar to another issue I mentioned. 
#8

Reply:
Where the second options can be satisfied, I think that's another problem.